### PR TITLE
Feat: Implements SlideOver Component

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?
- This PR attempts to implements SlideOver Component using[ FastStore's Modal](https://faststoreui.netlify.app/?path=/story/molecules-modal--modal) component as base

- [x] Replaces Filter's Modal for SlideOver

## How it works? 
|direction = 'leftSide' | 
|-|
leftside

|direction = 'rightSide'|
|-|
rightSide

## How to test it?
- Go to [preview link]( https://deploy-preview-262--basestore.netlify.app/office/)
- On the mobile resolution, tap on `Filters` 

## References
- [FSSS-148 SlideOver Spike](https://vtex-dev.atlassian.net/jira/software/projects/FSSS/boards/632?selectedIssue=FSSS-148)

- https://reactjs.org/docs/events.html#transition-events

Thanks @lucasfp13 for helping me to handle the callbacks functions 🙏

### CODE REVIEW

File: src/components/ui/SlideOver/SlideOver.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    <UIModal
+      isOpen={isOpen}
+      onDismiss={(e) => {
+        e.preventDefault()
+        handleClose()
+      }}
+      data-slide-over
+      data-slide-over-direction={direction}
+      data-slide-over-size={size}
+      data-slide-over-state={fadeType}
```
Comment (hellofanny) [inline src/components/ui/SlideOver/SlideOver.tsx:70]: fade?

File: src/components/ui/SlideOver/slide-over.scss [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+[data-slide-over-direction="rightSide"][data-slide-over-state="in"] {
+  transform: translateX(0);
+  transition: transform .25s ease-in;
+}
+
+[data-slide-over-direction="rightSide"][data-slide-over-state="out"] {
+  transform: translateX(100%);
+  transition: transform .25s ease-out;
+}
```
Comment (danzanzini) [inline src/components/ui/SlideOver/slide-over.scss:53]: Wow, Just learned a lot reading this file. Thanks 😄

**Review by danzanzini — CHANGES_REQUESTED** [review]
Great job, @hellofanny. The code is great.

I'm requesting changes because there is a problem in the behavior when selecting `View Results` in the filter. I think it should close the filter, but it closes and reopens.

https://user-images.githubusercontent.com/8497187/150543406-a7a4118e-a9d2-4520-a811-8d1e0f880fb6.mov

File: src/components/ui/SlideOver/SlideOver.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  ...otherProps
+}: SlideOverProps) => {
+  const [fadeType, setFadeType] = useState<FadeType>()
+
+  const handleClose = useCallback(() => {
+    setFadeType('out')
+  }, [])
+
+  useEffect(() => {
+    if (isOpen === undefined || isOpen) {
```
[the remark this answers is not in the document]
Comment (hellofanny) [reply src/components/ui/SlideOver/SlideOver.tsx:43]: Actually, I just need to verify `isOpen`. Lemme fix it!

File: src/components/search/Filter/Filter.tsx [hunk 4]
```diff
@@ -37,7 +36,7 @@ interface Props {
    * This function is called whenever the user hits "Escape", clicks outside
    * the filter modal or clicks in close button. (mobile only)
    */
-  onDismiss?: (event: MouseEvent | KeyboardEvent | undefined) => void
+  onDismiss?: () => void
```
[the remark this answers is not in the document]
Comment (lucasfp13) [reply src/components/search/Filter/Filter.tsx:28]: nice catch @danzanzini! It was fixed in f79de85

File: cypress/integration/plp.test.js [hunk 5]
```diff
@@ -20,26 +20,26 @@ describe('Search page Filters and Sorting options', () => {
...
-      .click()
+      .click({ force: true })
       .getById('filter-accordion-button')
       .first()
-      .click()
+      .click({ force: true })
       .getById('filter-accordion-panel-checkbox')
       .first()
-      .click()
+      .click({ force: true })
```
Comment (lucasfp13) [inline cypress/integration/plp.test.js:42]: Why did it need to have those forced clicks here?

File: src/components/ui/SlideOver/SlideOver.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState, useCallback } from 'react'
+import type { ReactNode, HTMLAttributes } from 'react'
+import { Modal as UIModal } from '@faststore/ui'
+
+import './slide-over.scss'
+
+type Direction = 'leftSide' | 'rightSide'
+type WidthSize = 'full' | 'partial'
+type FadeType = 'in' | 'out'
```
Comment (lucasfp13) [inline src/components/ui/SlideOver/SlideOver.tsx:9]: nicee! 💯

**Review by lucasfp13 — APPROVED** [review]
Great job! 🎉

File: cypress/integration/plp.test.js (hunk 5 above)
Comment (filipewl) [reply cypress/integration/plp.test.js:42]: Yeah, @hellofanny, try removing these added `force: true` because I think they aren't needed anymore.

**Review by filipewl — APPROVED** [review]
_Beautiful!_ Nice job @hellofanny and @lucasfp13. I think once we address [this comment](https://github.com/vtex-sites/base.store/pull/262/files?w=1#r790166671), this is good to be merged.

**Review by renatamottam — APPROVED** [review]
Amazing improvement, faaaaan 🥳

File: src/components/ui/SlideOver/SlideOver.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  }, [])
+
+  useEffect(() => {
+    if (isOpen) {
+      setFadeType('in')
+    }
+  }, [isOpen])
+
+  useEffect(() => {
+    if (handleClose) {
```
Comment (eduardoformiga) [inline src/components/ui/SlideOver/SlideOver.tsx:49]: I'm not sure if we need this `if` here since `handleClose` is a `function` and is always `true`. 👀

**Review by eduardoformiga — COMMENTED** [review]
✅ Amazing Job!  🪄 👏 

I parachuted here just now. I left some comments (not important).

File: src/components/ui/SlideOver/slide-over.scss [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+[data-slide-over-size="full"] {
+  width: 100vw;
+}
+
+[data-slide-over-size="partial"] {
+  --gap-out: 48px;
+
+  width: calc(100vw - var(--gap-out));
```
[the remark this answers is not in the document]
Comment (hellofanny) [reply src/components/ui/SlideOver/slide-over.scss:24]: Back then, I also suggested %, but flip (design team) said this should be a fixed gap.

File: src/components/ui/SlideOver/SlideOver.tsx (hunk 7 above)
Comment (hellofanny) [reply src/components/ui/SlideOver/SlideOver.tsx:49]: I think you are right! 🙈

**Pull request merged**